src/main/java/carpet/script/annotation/Locator.java

```diff
@@ -0,0 +1,270 @@
+package carpet.script.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.reflect.AnnotatedType;
+import java.util.Iterator;
+
+import org.apache.commons.lang3.NotImplementedException;
+
+import carpet.script.Context;
+import carpet.script.argument.Argument;
+import carpet.script.argument.BlockArgument;
+import carpet.script.argument.FunctionArgument;
+import carpet.script.argument.Vector3Argument;
+import carpet.script.bundled.Module;
+import carpet.script.value.BlockValue;
+import carpet.script.value.FunctionValue;
+import carpet.script.value.Value;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * <p>Class that holds annotation for {@link Argument} locators to be used in Scarpet functions.</p>
+ * 
+ * <p>Note: Nothing in this class has been implemented at this point in time, since it requires Carpet changes. There is an implementation for
+ * Locator.Block working, but not implemented.</p>
+ */
+public interface Locator
+{
+    /**
+     * <p>Represents that the annotated argument must be gotten by passing the arguments in there into a {@link BlockArgument} locator.</p>
+     * 
+     * <p>Must be used in either {@link BlockValue} or {@link BlockArgument} parameters</p>
+     */
+    @Documented
+    @Retention(RUNTIME)
+    @Target({ PARAMETER, TYPE_USE })
+    @Deprecated // Not implemented, although implementation available
```

Comment:
Why are you deprecating it? Are you hoping it will be merged with the general block locator using that instead?